### PR TITLE
Support compression in gRPC transactions

### PR DESCRIPTION
**Pull request by tanmayv25**
Uses per call compression only for inference requests. Using compression for non-infer related requests is not a good idea.

client-side compression:
`I0316 12:05:30.316294326    7617 message_compress_filter.cc:265] Compressed[deflate] 204 bytes vs. 102 bytes (50.00% savings)`

server-side compression:
`I0316 12:50:35.188782589   10734 message_compress_filter.cc:265] Compressed[deflate] 187 bytes vs. 99 bytes (47.06% savings)`

performance for sure seems to suffer. The numbers are collected on RN50 saved model:

```
No Compression: Concurrency: 1, throughput: 69 infer/sec, latency 14488 usec
Deflate: Concurrency: 1, throughput: 27.8 infer/sec, latency 35990 usec
Gzip: Concurrency: 1, throughput: 27.8 infer/sec, latency 35957 usec
```

### CODE REVIEW

**Comment by CoderHam** [comment]
Can you try this with a large payload size to see if it makes a difference for large payloads?

File: src/clients/python/examples/simple_grpc_infer_client.py [hunk 1]
```diff
@@ -78,6 +78,15 @@
         required=False,
         default=None,
         help='File holding PEM-encoded certicate chain. Default is None.')
+    parser.add_argument(
+        '-C',
+        '--compression_algorithm',
```
Comment (CoderHam) [inline src/clients/python/examples/simple_grpc_infer_client.py:83]: Since this is just for GRPC maybe call it `--grpc-compression-algorithm` instead?

**Review by CoderHam — CHANGES_REQUESTED** [review]

File: src/clients/python/examples/simple_grpc_infer_client.py (hunk 1 above)
Comment (tanmayv25) [reply src/clients/python/examples/simple_grpc_infer_client.py:83]: The client is itself grpc hence I omitted grpc here... But I can make this change for uniformity with perf_analyzer ( and perf_analyzer in the future)

Comment (CoderHam) [reply src/clients/python/examples/simple_grpc_infer_client.py:83]: Yes, I agree that having uniform flags will help 😄

**Comment by tanmayv25** [comment]
@CoderHam More than the tensor size, we will see improvements with larger concurrency so that the time spent in compressing and decompressing the messages are overlapped. The above perf numbers are for RN50 model with input size ~4MB.

**Comment by tanmayv25** [comment]
For the add sub model at a higher concurrency we see similar perf as the compression/decompression time is pipelined:
```
Deflate: Concurrency: 50, throughput: 2457.2 infer/sec, latency 20346 usec
Gzip: Concurrency: 50, throughput: 2494.4 infer/sec, latency 20044 usec
None: Concurrency: 50, throughput: 2461.6 infer/sec, latency 20308 usec
```

**Review by CoderHam — APPROVED** [review]

**Pull request merged**